aspnetcore/fundamentals/servers/kestrel/endpoints.md

```diff
@@ -220,54 +220,180 @@ For example, the `Certificates:Default` certificate can be specified as:
+* Ensure that the socket is writeable by Nginx (for example, `chmod go+w /tmp/kestrel-test.sock`).
+
+### Dynamic port binding
+
+For information about dynamic port binding, see [Port 0](#port-0) earlier in this article.
+
+## Configure HTTPS
+
+Kestrel supports securing endpoints with HTTPS. The way you configure HTTPS depends on how endpoints are defined.
+
+If [URL prefixes](#configure-endpoints-with-urls) are used to define endpoints, HTTPS can be used only if a default certificate is provided in HTTPS endpoint configuration. For example, use <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions> configuration or a configuration file as shown [earlier in this article](#configure-certificates-in-appsettingsjson).
+
+For more information about configuring HTTPS, such as using configuration or code to specify SSL certificates and SSL/TLS protocols, see the following sections of this article:
+
+* [Configure certificates in appsettings.json](#configure-certificates-in-appsettingsjson)
+* [Configure client certificates in appsettings.json](#configure-client-certificates-in-appsettingsjson)
+* [Certificate sources](#certificate-sources)
+* [Configure HTTPS defaults in code](#configure-https-defaults-in-code)
+* [Configure certificates in code](#configure-certificates-in-code)
+* [Configure client certificates in code](#configure-client-certificates-in-code)
+* [Configure endpoints using Server Name Indication (SNI)](#configure-endpoints-using-server-name-indication)
+* [Connection middleware](#connection-middleware)
+* [Configure endpoint protocols](#configure-endpoint-protocols)
 
-### SNI in configuration
+To create a development certificate, use the [dev-certs tool](/dotnet/core/tools/dotnet-dev-certs). The tool is automatically installed when the [.NET SDK](/dotnet/core/sdk) is installed.
 
-Kestrel supports SNI defined in configuration. An endpoint can be configured with an `Sni` object that contains a mapping between host names and HTTPS options. The connection host name is matched to the options and they are used for that connection.
+The development certificate is available only for the user that generates the certificate. Some browsers require granting explicit permission to trust the local development certificate.
+
+Many ASP.NET Core project templates configure apps to run on HTTPS by default and include [HTTPS redirection and HSTS support](xref:security/enforcing-ssl).
+
+## Configure endpoints using Server Name Indication
+
+[Server Name Indication (SNI)](https://tools.ietf.org/html/rfc6066#section-3) can be used to host multiple domains on the same IP address and port. For SNI to function, the client sends the host name for the secure session to the server during the TLS handshake so that the server can provide the correct certificate. The client uses the furnished certificate for encrypted communication with the server during the secure session that follows the TLS handshake.
```

Comment:
```suggestion
[Server Name Indication (SNI)](https://tools.ietf.org/html/rfc6066#section-3) can be used to host multiple domains on the same IP address and port. You might want to do this to conserve resources by serving multiple sites from one server.

For SNI to function, the client sends the host name for the secure session to the server during the TLS handshake so that the server can provide the correct certificate. The client uses the furnished certificate for encrypted communication with the server during the secure session that follows the TLS handshake.
```